### ISSUE TITLE
"Error: Invalid argument type" when 2 types added to DB_DUMP_TARGET

### ISSUE THREAD

**Issue by egahmad** [databacker/mysql-backup#156]
I followed the instructions added whitespace between local and s3 destination but I get a wrong type error, here's my docker-compose

```yaml
backup:
    image: databack/mysql-backup
    container_name: databack-backup
    restart: always
    volumes:
      - /db:/db
      - type: volume
        source: databack-pre
        target: /scripts.d/pre-backup
      - type: volume
        source: databack-post
        target: /scripts.d/post-backup
    environment:
      - DB_SERVER=mysql
      - DB_USER=user
      - DB_PASS=pass
      - DB_PORT=3306
      - DB_DUMP_FREQ=1440
      - DB_DUMP_BEGIN=2330
      - DB_DUMP_TARGET=/db s3://dir/db
      - AWS_ACCESS_KEY_ID=key
      - AWS_SECRET_ACCESS_KEY=secret
    networks:
      - "db"
```

**Comment by deitch** [comment]
Agreed, it should work. Run it with debug enabled `DB_DUMP_DEBUG=true` and post the logs please. Make sure to go through it and be sure you don't leak any secrets.

**Comment by egahmad** [comment]
I got this
databack-backup   | + aws s3 cp /tmp/backups/db_backup_2021-02-16T19:49:35Z.tgz '/db s3://buvket/db/db_backup_2021-02-16T19:49:35Z.tgz'
databack-backup   |
databack-backup   | usage: aws s3 cp <LocalPath> <S3Uri> or <S3Uri> <LocalPath> or <S3Uri> <S3Uri>
databack-backup   | Error: Invalid argument type

**Comment by deitch** [comment]
That doesn't look right. It should be running that twice, once for each one.

Can you paste the whole debug output?

**Comment by ToshY** [comment]
@deitch The debug provided clearly shows that the targets are not split. I'm pretty certain it's due to [line 199 in entrypoint](https://github.com/databacker/mysql-backup/blob/b4449db1f076bc187273ea3b59124f5a40940bb1/entrypoint#L199), which doesn't split the targets.

**Comment by deitch** [comment]
Why wouldn't it? It isn't quoted.

```console
$ export ABC="1 2"
$ for i in $ABC; do
> echo $i
> done
1
2
```

It is possible that docker compose is doing something when constructing it., but straight docker does not:

```console
$ docker run -it --rm --env ABC="1 2" alpine:3.14 sh -c 'for i in $ABC; do echo $i; done'
1
2

**Comment by deitch** [comment]
This sounds odd, but what if you try quoting it, as in

```yaml
    environment:
      - DB_DUMP_TARGET="/db s3://dir/db"
```

**Comment by ToshY** [comment]
@deitch Yes I spoke to soon, sorry about that. I thought I tested with quotes as well but apparently didn't. In hindsight it makes sense. The debug log now shows going through both targets, and dump to both as well. 

Maybe a minor addition to the docs for docker-compose with regards to multiple targets would be nice? I could submit a PR myself if you'd like.

**Comment by deitch** [comment]
Glad we figured it out. Sure, a PR is appreciated.

**Comment by deitch** [comment]
Closed via #175
